### PR TITLE
Prevent blank username and invalid password

### PR DESCRIPTION
**Pull request by Sasha45**
Forbid empty usernames
Forbid usernames under two characters
Forbid usernames with non-word characters
Forbid PINs under 4 characters
Forbid PINs with non-numeric characters

### CODE REVIEW

**Comment by Sasha45** [comment]
I expect merge conflicts with #34 :(    Whichever one gets approved second will have to resolve those.

**Review by AlejandroM2000 — DISMISSED** [review]

File: source/tests/PuppTests.test.js [hunk 1]
```diff
@@ -59,6 +59,97 @@ describe('basic navigation for BJ', () => {
     });
     */
 
+    //Login tests
+
+    /*
+    it('LoginTest1: no username', async () => {
```
Comment (k9kathyli) [inline source/tests/PuppTests.test.js:65]: nit: make test names more descriptive of what they're doing

**Review by k9kathyli — CHANGES_REQUESTED** [review]

**Comment by k9kathyli** [comment]
- [x] Straightforward implementation
- [x] Good design
- [x] Code looks correct
- [x] UI looks as intended and fits guidelines
- [] Adequate test coverage
- [x] Follows coding conventions 
- [] Code is commented/documented well

File: source/tests/PuppTests.test.js [hunk 2]
```diff
@@ -59,6 +59,97 @@ describe('basic navigation for BJ', () => {
     });
     */
 
+    //Login tests
```
[the remark this answers is not in the document]
Comment (Sasha45) [reply source/tests/PuppTests.test.js:62]: image

**Comment by Sasha45** [comment]
Proof of passing tests
image

**Review by k9kathyli — APPROVED** [review]

**Review by AlejandroM2000 — APPROVED** [review]

**Pull request merged**